### PR TITLE
refactor(core) Remove enums from event-dispatch.

### PR DESCRIPTION
**Pull request by iteriani**
These cause optimization issues in external.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/core/primitives/event-dispatch/src/accessibility.ts [hunk 1]
```diff
@@ -6,39 +6,41 @@
...
-
-  /**
-   * A property added to a dispatched event that had the MAYBE_CLICK_EVENTTYPE
-   * event-type but could not be used as a click. The dispatcher sets this
-   * property before it retriggers the event and it signifies that the event
-   * contract should not look at CLICK actions for KEYDOWN events.
-   */
-  SKIP_A11Y_CHECK = 'a11ysc',
-}
+export const Attribute = {
```
Comment (rahatarmanahmed) [inline packages/core/primitives/event-dispatch/src/accessibility.ts:34]: I think we can delete this now

(also these aren't really Attributes)

**Review by rahatarmanahmed — APPROVED** [review]

File: packages/core/primitives/event-dispatch/src/accessibility.ts (hunk 1 above)
Comment (iteriani) [reply packages/core/primitives/event-dispatch/src/accessibility.ts:34]: Is it ok to do in a followup change?

Comment (rahatarmanahmed) [reply packages/core/primitives/event-dispatch/src/accessibility.ts:34]: yeah

File: packages/core/primitives/event-dispatch/src/restriction.ts [hunk 2]
```diff
@@ -11,5 +11,5 @@
  */
 
 export enum Restriction {
-  I_AM_THE_JSACTION_FRAMEWORK = 1,
+  I_AM_THE_JSACTION_FRAMEWORK,
```
Comment (AndrewKushnir) [inline packages/core/primitives/event-dispatch/src/restriction.ts:14]: Why did we change a value here?

**Review by AndrewKushnir — APPROVED** [review]

**Review by thePunderWoman — APPROVED** [review]
reviewed-for: public-api

File: packages/core/primitives/event-dispatch/src/restriction.ts (hunk 2 above)
Comment (iteriani) [reply packages/core/primitives/event-dispatch/src/restriction.ts:14]: didnt seem necessary

**Review by AndrewKushnir — APPROVED** [review]

**Comment by iteriani** [comment]
[TGP](https://fusion2.corp.google.com/presubmit/631531787/OCL:631531787:BASE:631861196:1715191768063:c9d5115d;groups=PossiblyNewlyFailing/targets)

**Review by thePunderWoman — APPROVED** [review]
reviewed-for: size-tracking

**Comment by iteriani** [comment]
TESTED=[TGP](https://fusion2.corp.google.com/presubmit/631531787/OCL:631531787:BASE:631861196:1715191768063:c9d5115d;groups=PossiblyNewlyFailing/targets)

**Comment by AndrewKushnir** [comment]
@iteriani this PR conflicts with the `main` branch (a golden file needs an update) after merging https://github.com/angular/angular/pull/55587. Could you please rebase and remove the "cleanup" label from this PR when you get a chance?

**Comment by atscott** [comment]
This PR was merged into the repository by commit 3068de43692dff76a9c9b5665a2dc17cc4ed1190.

**Pull request closed without merging**